lib/utils/createConfig.js

```diff
@@ -47,11 +47,11 @@ function createConfig(config, argv, { port }) {
     }
   }
 
-  if (!options.filename) {
+  if (!options.filename && firstWpOpt.output && firstWpOpt.output.filename) {
```

Comment:
Set only when configuration specify (otherwise it was `undefined`, what doesn't have sense)

Reply:
Same logic for other arguments too